test/cli/harness/harness.go

```diff
@@ -20,42 +19,35 @@ type Harness struct {
 	Runner    *Runner
 	NodesRoot string
 	Nodes     Nodes
+	Log       *TestLogger
+	T         *testing.T
 }
 
-// TODO: use zaptest.NewLogger(t) instead
-func EnableDebugLogging() {
-	err := logging.SetLogLevel("testharness", "DEBUG")
-	if err != nil {
-		panic(err)
+// New constructs a harness that cleans up after the given test is done.
+func New(t *testing.T, options ...func(h *Harness)) *Harness {
+	log := NewTestLogger(t)
```

Comment:
Should we add t.Helper() here?

Reply:
Could you help me understand when t.Helper() is useful?  AFAIU it's just for slightly nicer stack traces?

Reply:
Oh so it is for test reporting tools so that they show you the right line?

Reply:
AFAIU t.Helper() is just for the calling function, New() in this case, any reason why that is particularly special here?

Reply:
> Oh so it is for test reporting tools so that they show you the right line?

Right https://pkg.go.dev/testing#T.Helper

Reply:
> any reason why that is particularly special here?

Just I saw that you added this new method that is used on every test. Marking this method as a helper will improve dev experience when some of these tests fail.